src/include/target.h

```diff
@@ -30,6 +30,12 @@
 #include <stdint.h>
 #include <sys/types.h>
 
+#if defined(_MSC_VER)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+typedef unsigned int mode_t;
```

Comment:
> rather reaches into UCRT/Windows internal headers

Hi @dragonmux, this is not correct. The header is documented here: https://learn.microsoft.com/en-us/windows/win32/winprog/windows-data-types#ssize_t and it's prescribed as the one to include. 

As for `mode_t`, I'd suggest to have a look here: https://github.com/mingw-w64/mingw-w64/blob/7ea930ba381ace6f19f7cfa59596f6fe49949aa6/mingw-w64-headers/crt/sys/types.h#L74, I'm not sure if it must be 32-bit.

Reply:
Ooo, thank you for that - we'll amend the suggestion then:
```suggestion
#include <BaseTsd.h>
typedef SSIZE_T ssize_t;
typedef int32_t mode_t;
```

The definition we're using for `mode_t` here is from substrate and was chosen because it's what both the C standard prescribes and the correct/compatible type for the Windows C runtime (they just use the raw type rather than a typedef). C made it signed (prescribed `int` specifically).